interpreter/CMakeLists.txt

```diff
@@ -134,24 +160,243 @@ set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "")
+      #    AND NOT EXISTS ${LLVM_LIBRARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}gtest${CMAKE_STATIC_LIBRARY_SUFFIX}
+      #    AND EXISTS ${UNITTEST_DIR}/CMakeLists.txt)
+      #  add_subdirectory(${UNITTEST_DIR} utils/unittest)
+      #endif()
+    else()
+      # Seek installed Lit.
+      find_program(LLVM_LIT
+                   NAMES llvm-lit lit.py lit
+                   PATHS "${ROOT_LLVM_MAIN_SRC_DIR}/utils/lit"
+                   DOC "Path to lit.py")
+    endif()
+
+    if(LLVM_LIT)
+      # Define the default arguments to use with 'lit', and an option for the user
+      # to override.
+      set(LIT_ARGS_DEFAULT "-sv")
+      if (MSVC OR XCODE)
+        set(LIT_ARGS_DEFAULT "${LIT_ARGS_DEFAULT} --no-progress-bar")
+      endif()
+      set(LLVM_LIT_ARGS "${LIT_ARGS_DEFAULT}" CACHE STRING "Default options for lit")
+      set(LIT_COMMAND "${LLVM_LIT}" CACHE STRING "Path to the LLVM LIT.")
+
+      # On Win32 hosts, provide an option to specify the path to the GnuWin32 tools.
+      if( WIN32 AND NOT CYGWIN )
+        set(LLVM_LIT_TOOLS_DIR "" CACHE PATH "Path to GnuWin32 tools")
+      endif()
+    else()
+      set(LLVM_INCLUDE_TESTS OFF)
+    endif()
+  endif()
+
+  set( CLANG_BUILT_STANDALONE 1 )
+  set(BACKEND_PACKAGE_STRING "LLVM ${LLVM_PACKAGE_VERSION}")
+
+  add_subdirectory(llvm/src/tools/clang)
+endif(builtin_llvm)
+
+#---Export the include directories------------------------------------------------------------------
+if (builtin_clang)
+  set(CLANG_INCLUDE_DIRS
+    ${CMAKE_SOURCE_DIR}/interpreter/llvm/src/tools/clang/include
+    ${CMAKE_BINARY_DIR}/interpreter/llvm/src/tools/clang/include
+    CACHE STRING "Clang include directories.")
+endif()
+
+if(builtin_llvm)
+  set(LLVM_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/interpreter/llvm/src/include
+    ${CMAKE_BINARY_DIR}/interpreter/llvm/src/include
+    CACHE STRING "LLVM include directories."
+    )
+  #---Set into parent scope LLVM_VERSION --------------------------------------------------------------
+  set(llvmconfigfile "${CMAKE_CURRENT_SOURCE_DIR}/llvm/src/CMakeLists.txt")
+  file(READ ${llvmconfigfile} _filestr)
+  foreach(versionpart LLVM_VERSION_MAJOR LLVM_VERSION_MINOR LLVM_VERSION_PATCH)
+    if("${_filestr}" MATCHES "set[(](${versionpart}[^)]+)")
+      string(REGEX REPLACE " *${versionpart} ([^)]+).*" "\\1" ${versionpart} "${CMAKE_MATCH_1}")
+    else()
+      MESSAGE(FATAL_ERROR "Cannot extract LLVM version number (${versionpart}) from ${llvmconfigfile}")
+    endif()
+  endforeach()
+  set(LLVM_VERSION "${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}" PARENT_SCOPE)
+else()
+  set(LLVM_INCLUDE_DIRS ${LLVM_MAIN_INCLUDE_DIR}
+    CACHE STRING "System LLVM include directories."
+    )
+  # We cannot include LLVMConfig.cmake here. We reply on builtin clang to discover the available
+  # llvm version and set it in clang version.
+  # FIXME: This is a very ugly workaround. Is there a better way to do this?
```

Comment:
Maybe you should consider addressing this and other `FIXME` comments before merging this.

Reply:
That's a reminder that we are being suboptimal here. I am unaware of how to fix it at that point in time, even if I had the spare cycles.

Reply:
Fine, but then please let me take a look at this (since I am already working on improving our current system for externals/builtins) to try to find a better solution instead of merging something suboptimal.

Reply:
I'd prefer to merge this because it is blocking a few things. Then once you find a better solution feel free to change this code there.

Reply:
For this one time, sure. Go ahead. But I will not accept merging of hacky changes to the build system in the future anymore.